src/listeners/MovementListener.java

```diff
@@ -1,265 +1,245 @@
-                                                   int columnClicked) {
-        return gui.boardPieces[rowClicked][columnClicked] != null
-                && this.selectedPiece == null
-                && this.secondSelectedPiece == null;
-    }
-
-    private boolean clickOnBoard(int rowClicked, int columnClicked) {
-        return rowClicked <= 7 && rowClicked >= 0 && columnClicked <= 7
-                && columnClicked >= 0;
-    }
-
-    private boolean checkForPush(int rowClicked, int columnClicked) {
-        if (this.secondSelectedPiece.getRow() + 1 == rowClicked
-                && this.secondSelectedPiece.getColumn() == columnClicked) {
-            return true;
-        }
-        if (this.secondSelectedPiece.getRow() - 1 == rowClicked
-                && this.secondSelectedPiece.getColumn() == columnClicked) {
-            // numMoves-=2;
-            return true;
-        }
-        if (this.secondSelectedPiece.getRow() == rowClicked
-                && this.secondSelectedPiece.getColumn() + 1 == columnClicked) {
-            // numMoves-=2;
-            return true;
-        }
-        if (this.secondSelectedPiece.getRow() == rowClicked
-                && this.secondSelectedPiece.getColumn() - 1 == columnClicked) {
-            // numMoves-=2;
-            return true;
-        }
-        return false;
-    }
-
-    private boolean checkForPull(int rowClicked, int columnClicked) {
-        if (this.selectedPiece.getRow() + 1 == rowClicked
-                && this.selectedPiece.getColumn() == columnClicked) {
-            // numMoves-=2;
-            return true;
-        }
-        if (this.selectedPiece.getRow() - 1 == rowClicked
-                && this.selectedPiece.getColumn() == columnClicked) {
-            // numMoves-=2;
-            return true;
-        }
-        if (this.selectedPiece.getRow() == rowClicked
-                && this.selectedPiece.getColumn() + 1 == columnClicked) {
-            // numMoves--;
-            return true;
-        }
-        if (this.selectedPiece.getRow() == rowClicked
-                && this.selectedPiece.getColumn() - 1 == columnClicked) {
-            // numMoves--;
-            return true;
-        }
-        return false;
-    }
+	private final Game game;
+	ImagePanel selectedPiece;
+	Coordinate selectedPieceCoord;
+	ImagePanel secondSelectedPiece;
+	Coordinate secondSelectedPieceCoord;
+	private GUI gui;
+
+	public MovementListener(GUI gui) {
+		this.game = gui.game;
+		this.gui = gui;
+		this.selectedPiece = null;
+		this.secondSelectedPiece = null;
+	}
+
+	@Override
+	public void mouseClicked(MouseEvent e) {
+		// Not needed
+	}
+
+	@Override
+	public void mouseReleased(MouseEvent e) {
+		// Not needed
+	}
+
+	@Override
+	public void mouseEntered(MouseEvent e) {
+		// Not needed
+	}
+
+	@Override
+	public void mouseExited(MouseEvent e) {
+		// Not needed
+	}
+
+	@Override
+	public void mousePressed(MouseEvent e) {
```

Comment:
this method is a bit long, you might want to consider making some of it into additional methods